doc/en/user/source/tutorials/GetFeatureInfo/geojson.rst

```diff
@@ -0,0 +1,206 @@
+.. _tutorials_getfeatureinfo_geojson:
+
+GeoJSON output format
+======================
+
+The default GeoJSON output uses the WFS GeoJSON  encoding mechanism, producing a fixed output, it is however possible to customize the output using FreeMarker templates.
+GeoServer will lookup for json templates following the same rules defined for the hmtl output, but the template files have to be named appending ``_json`` to the usual name, as below:
+
+* ``header_json.ftl``
+* ``content_json.ftl``
+* ``footer_json.ftl``
+
+Moreover, unlike the html case, all three template files must be provided.
+In case of a multi-layer request GeoServer will act in the following way:
+
+* content template will be searched up following the usual rules;
+* since there are no default templates for GeoJSON, header and footer will be looked up in the ``templates`` directory;
+* features with no content template will be encoded with the normal GeoJSON encoding, along with the customized ones.
```

Comment:
Seems to suggest that in a mixed templated/templated case the templates would still be used for the collections having the template. However, looking at the code, it seems the templates are used only if all collections have them instead. Please clarify.

Reply:
The method to check the presence of needed templates return true if at least one content template is found https://github.com/geoserver/geoserver/pull/4181/commits/7b1450e0172c8604de5e09c33423a4e455d69aec#diff-23918ce152856383ab9ceb08815749fbR246. But probably I had a bad idea in keeping the logic for both format in the templateManager. I made it an abstract class, so that each format has its own custom implementation, now the functioning should be more visible https://github.com/geoserver/geoserver/pull/4181/commits/9826f9d6477444add511c2f0e6e06685e23c9d12#diff-91952e39c8f5fd482e5bdf5e637d9967R90